okio/src/commonTest/kotlin/okio/AbstractFileSystemTest.kt

```diff
@@ -131,6 +131,36 @@ abstract class AbstractFileSystemTest(
     source.close()
   }
 
+  @Test
+  fun readPath() {
+    val path = base / "read-path"
+    val string = "hello, read with a Path"
+    path.writeUtf8(string)
+
+    val byteCount = fileSystem.read(path) {
+      assertEquals("hello", readUtf8(5))
+      assertEquals(", read with ", readUtf8(12))
+      assertEquals("a Path", readUtf8())
+      return@read "success"
+    }
+    assertEquals("success", byteCount)
+  }
+
+  @Test
+  fun readString() {
+    val path = base / "read-string"
+    val string = "hello, read with a String"
+    path.writeUtf8(string)
+
+    val byteCount = fileSystem.read(path.toString()) {
```

Comment:
why byteCount? result? It's a String.

Reply:
whoops! Fixed. Just wanted to make sure the return value went through.